src/core/bpf-firewall.c

```diff
@@ -657,25 +657,18 @@ static int attach_custom_bpf_progs(Unit *u, const char *path, int attach_type, S
         assert(u);
 
         set_clear(*set_installed);
+        *set_installed = set_new(&bpf_program_hash_ops);
```

Comment:
I think you want `set_ensure_allocated()` instead of `set_new()`. The former checks if the set is already allocated. `set_clear()` doesn't free the set so the old `set_installed` could be leaked when you overwrite it with this new one.